xchem_db/models.py

```diff
@@ -175,32 +451,53 @@ class Meta:
-    crystal_name = models.OneToOneField(Crystal, on_delete=models.CASCADE, unique=True)  # changed to foreign key
-    data_collection_visit = models.TextField(blank=True, null=True)
+
+    #crystal_name = models.OneToOneField(Crystal, on_delete=models.CASCADE, unique=True)  # changed to foreign key
+    compound = models.OneToOneField(SoakDBCompound, on_delete=models.CASCADE, unique=True)  #changed to allow cocktails
+    #to access crystal_name now: self.compound.crystal
+        
+    data_collection_visit = models.CharField(max_length=64, blank=True, null=True)
     expr_conc = models.FloatField(blank=True, null=True)
-    harvest_status = models.TextField(blank=True, null=True)
-    library_name = models.TextField(blank=True, null=True)
-    library_plate = models.TextField(blank=True, null=True)
-    mounting_result = models.TextField(blank=True, null=True)
-    mounting_time = models.TextField(blank=True, null=True)
-    soak_status = models.TextField(blank=True, null=True)
-    soak_time = models.TextField(blank=True, null=True)
-    soak_vol = models.FloatField(blank=True, null=True)
-    solv_frac = models.FloatField(blank=True, null=True)
-    stock_conc = models.FloatField(blank=True, null=True)
-    visit = models.TextField(blank=True, null=True)
+    harvest_status = models.CharField(max_length=64, blank=True, null=True)
+    mounting_result = models.CharField(max_length=64, blank=True, null=True)
+    mounting_time = models.CharField(max_length=64, blank=True, null=True)
+    visit = models.CharField(max_length=64, blank=True, null=True)
+
+    #new attributes
+    batch = models.ForeignKey(Batch, blank=True, null=True, on_delete=models.PROTECT) #null for solvent testing
+    solvent_data = models.ForeignKey(SolventTestingData, blank=True, null=True, on_delete=models.PROTECT) #null for compound screen
+    puck = models.CharField(max_length=100, blank=True, null=True)
+    position = models.CharField(max_length=100, blank=True, null=True)
+    pin_barcode = models.CharField(max_length=100, blank=True, null=True)
+    arrival_time = models.DateTimeField(blank=True, null=True)
+    mounted_timestamp = models.DateTimeField(blank=True, null=True)
+    ispyb_status = models.CharField(max_length=100, blank=True, null=True)
+    
 
     class Meta:
         if os.getcwd() != '/dls/science/groups/i04-1/software/luigi_pipeline/pipelineDEV':
             app_label = 'xchem_db'
         db_table = 'lab'
 
+'''
+#(requires installing json field:
+#pip install jsonfield)
+from jsonfield import JSONField
+
+class SolventAutofill(models.Model):
+    #patterns of typical parameters used in solvent testing; used for auto-filling data
+    
+    name = models.CharField(max_length=100)
+    solvents = JSONField() #array of strings
+    solvent_concentrations = JSONField() #array of floats
+    cryo = models.BooleanField()
+    comment = models.textField()
+'''
+
+
+#ONLY OLD UNCHANGED MODELS BELOW
```

Comment:
Add jsonfield to requirements on xchemdb and Add the class.

Reply:
Or figure out what is happening when XCDB migrates.

Reply:
Array fields specific to Postgres - need to work through json field stuff?